### PR TITLE
fix: check the error chain when validating if it's x509 error

### PR DESCRIPTION
**Pull request by lubronzhan**
## Description

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

Check the error chain when checking if error is x509 related. This should be compatible with both golang 1.20 and 1.19

Closes: #https://github.com/vmware/govmomi/issues/3174

## Type of change

Please mark options that are relevant:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] This change requires a documentation update
- [ ] Build related change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide
instructions so we can reproduce. If applicable, please also list any relevant
details for your test configuration.

- [x] Test Description 1
Compiles CPI with 1.20 and configures CPI with tlsthumbprint, and the error doesn't appear.
- [ ] Test Description 2

## Checklist:

- [x] My code follows the `CONTRIBUTION`
  [guidelines](https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md) of
  this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

**Comment by sbueringer** [comment]
cc @dougm

**Comment by lubronzhan** [comment]
I think Doug is on PTO this week, he should be able to review this next week

**Comment by sbueringer** [comment]
Perfect!

File: vim25/soap/error.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ import (
 	"reflect"
 	"strings"
 
+	"github.com/pkg/errors"
```
Comment (dougm) [inline vim25/soap/error.go:26]: We should be using the [stdlib errors](https://pkg.go.dev/errors) package instead, right?

**Review by dougm — COMMENTED** [review]
Thanks @lubronzhan 
The [commit message](https://github.com/vmware/govmomi/blob/main/CONTRIBUTING.md#format-of-the-commit-message) is used to generate change log, auto-close issues, etc., can you please update?  Currently is:
```
issue-3174: check the error chain when validating if it's
 x509 error
```

Should include `fix:` prefix and `Closes: #NNNN`:

```
fix: check the error chain when validating x509 error

Closes: #3174
```

File: vim25/soap/error.go (hunk 1 above)
Comment (lubronzhan) [reply vim25/soap/error.go]: Thanks for the comment. I updated accordingly

**Review by dougm — APPROVED** [review]
Thanks @lubronzhan 
We can merge when CI passes (snapshot release failed, sometimes will timeout)

**Comment by sbueringer** [comment]
@dougm Newbie question (entirely new to CAPV / govmomi). How quickly could we get a release which includes this fix? :)

**Comment by sbueringer** [comment]
Perfect.

Thank you very much!!

**Pull request merged**